src/generate.rs

```diff
@@ -356,16 +355,71 @@ fn register_block(registers: &[Register], defs: &Defaults) -> Result<Tokens> {
     let mut i = 0;
     // offset from the base address, in bytes
     let mut offset = 0;
-    for register in util::expand(registers) {
-        let pad = if let Some(pad) = register.offset.checked_sub(offset) {
+    let mut registers_expanded = vec![];
+
+    for register in registers {
+        match *register {
```

Comment:
could you add a note here indicating that we'll only map a register array to an array type if there are no holes and each register in the array is 32 bits wide? Why is the 32 bits check required, by the way? Other registers arrays will be simply unrolled / expanded.